### ISSUE TITLE
Batchnorm in network_kernels.cu

### ISSUE THREAD

**Issue by martin-91x** [AlexeyAB/darknet#3524]
Hi,
I'm currently playing around a bit with darknet (printing additional stuff and so on) and I noticed that a `printf("BatchNorm: %d\n", l.batch_normalize);` in the main loop of the _forward_network_gpu_ in _network_kernels.cu_ outputs always 0.

However, adding the same line in _convolutional_layer.c_ inside the _make_convolutional_layer_ function leads to outputs corresponding to the *.cfg file.

Now I'm wondering why this is happening.

Best,
Martin

**Comment by AlexeyAB** [comment]
@martin-91x Hi,

For Detection only the batchnorm+conv are fused to a single conv-layer: https://github.com/AlexeyAB/darknet/blob/d4402d29c2845a9aef97f9394dc619986ffbdf0c/src/detector.c#L1284

Read: https://machinethink.net/blog/object-detection-with-yolo/

**Comment by martin-91x** [comment]
Hi,
thank you for your fast reply. 
Yeah, I know that the layers are fused - I just thought that the fused layer still has `batch_normalize=1` (and I missed the function call you mentioned). Everything's clear now.

Thank you.
